File: Linked_Lists/Easy/206.ReverseLinkedList/206.c
#include <stdio.h>

// Definition for singly-linked list.
struct ListNode {
    int val;
    struct ListNode *next;
};

struct ListNode* reverseList(struct ListNode* head){
    struct ListNode *previous = NULL;
    struct ListNode *next = NULL;
    while(head != NULL){
        next = head->next;
        head->next = previous;
        previous = head;
        head = next;
    }
    return previous;
}

File: Linked_Lists/Easy/206/206.js

// Definition for singly-linked list.
function ListNode(val, next) {
    this.val = (val===undefined ? 0 : val)
    this.next = (next===undefined ? null : next)
}

/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    let previous = null;
    let next = null;
    while(head != null){
        next = head.next;
        head.next = previous;
        previous = head;
        head = next
    }
    return previous
};